raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py

```diff
@@ -128,100 +128,70 @@ def test_is_safe_to_wait():
+        {'our_state': {'balance': amount}},
+        {'our_state': {'balance': 0}},
+        {'our_state': {'balance': amount}},
+    ])
 
     # the first available route should be used
-    available_routes = [factories.route_from_channel(channel1)]
     chosen_channel = mediator.next_channel_from_routes(
-        available_routes,
-        channel_map,
+        channels.get_routes(0),
+        channels.channel_map,
         amount,
         timeout_blocks,
     )
-    assert chosen_channel.identifier == channel1.identifier
+    assert chosen_channel.identifier == channels[0].identifier
 
     # additional routes do not change the order
-    available_routes = [
-        factories.route_from_channel(channel1),
-        factories.route_from_channel(channel2),
-    ]
     chosen_channel = mediator.next_channel_from_routes(
-        available_routes,
-        channel_map,
+        channels.get_routes(0, 1),
+        channels.channel_map,
         amount,
         timeout_blocks,
     )
-    assert chosen_channel.identifier == channel1.identifier
+    assert chosen_channel.identifier == channels[0].identifier
 
-    available_routes = [
-        factories.route_from_channel(channel3),
-        factories.route_from_channel(channel1),
-    ]
     chosen_channel = mediator.next_channel_from_routes(
-        available_routes,
-        channel_map,
+        channels.get_routes(2, 0),
+        channels.channel_map,
         amount,
         timeout_blocks,
     )
-    assert chosen_channel.identifier == channel3.identifier
+    assert chosen_channel.identifier == channels[2].identifier
 
     # a channel without capacity must be skipped
-    available_routes = [
-        factories.route_from_channel(channel2),
-        factories.route_from_channel(channel1),
-    ]
     chosen_channel = mediator.next_channel_from_routes(
-        available_routes,
-        channel_map,
+        channels.get_routes(1, 0),
+        channels.channel_map,
         amount,
         timeout_blocks,
     )
-    assert chosen_channel.identifier == channel1.identifier
+    assert chosen_channel.identifier == channels[0].identifier
 
 
 def test_next_route_reveal_timeout():
     """ Routes with a larger reveal timeout than timeout_blocks must be ignored. """
-    amount = 10
-    balance = 20
     timeout_blocks = 10
 
-    channel1 = factories.make_channel(our_balance=balance, reveal_timeout=timeout_blocks * 2)
-    channel2 = factories.make_channel(our_balance=balance, reveal_timeout=timeout_blocks + 1)
-    channel3 = factories.make_channel(our_balance=balance, reveal_timeout=timeout_blocks // 2)
-    channel4 = factories.make_channel(our_balance=balance, reveal_timeout=timeout_blocks)
-
-    channel_map = {
-        channel1.identifier: channel1,
-        channel2.identifier: channel2,
-        channel3.identifier: channel3,
-        channel4.identifier: channel4,
-    }
+    channels = factories.make_channel_set([
+        {'reveal_timeout': timeout_blocks * 2},
```

Comment:
The same comment above applies if you're willing to implement `ChannelProperties`